src/Util/Lifecycle/State/PosStateService.php

```diff
@@ -48,6 +52,14 @@ public function __construct(
 
     public function addPosSalesChannelType(Context $context): void
     {
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('typeId', SwagPayPal::SALES_CHANNEL_TYPE_POS));
+        $salesChannels = $this->salesChannelRepository->searchIds($criteria, $context);
+
+        if ($salesChannels->getTotal() > 0) {
+            return;
+        }
```

Comment:
I don't understand why this is here. Why should the Sales Channel Type not be upserted?